arraycontext/container/__init__.py

```diff
@@ -145,6 +147,29 @@ class ArrayContainer(Protocol):
     # that are container-typed.
 
 
+class ArithArrayContainer(ArrayContainer, Protocol):
```

Comment:
Does this need to be a subclass of `ArrayContainer`? Not sure how subclassing protocols is supposed to work..

Reply:
- `ArrayContainer` is an empty protocol, so functionally the inheritance is a no-op, I think.
- An array container with arithmetic *is* an array container, so the inheritance "says the right thing" to me.

IDK. If you feel strongly, I'd be happy to remove it.

Reply:
No, that's perfectly fine!